src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -3609,29 +3609,37 @@ static void writeBeam(XmlWriter& xml, ChordRest* const cr, Beam* const b)
         LOGD("Beam::writeMusicXml(): cannot find ChordRest");
         return;
     }
-    int blp = -1;   // beam level previous chord
-    int blc = -1;   // beam level current chord
-    int bln = -1;   // beam level next chord
-    BeamMode bmc = BeamMode::AUTO; // beam mode current chord
-    BeamMode bmn = BeamMode::AUTO; // beam mode next chord
-    // find beam level previous chord
+    int blp = -1;   // beam level previous chord or rest
+    int blc = -1;   // beam level current chord or rest
+    int bln = -1;   // beam level next chord or rest
+    BeamMode bmc = BeamMode::AUTO; // beam mode current chord or rest
+    BeamMode bmn = BeamMode::AUTO; // beam mode next chord or rest
+    // find beam level previous chord or rest
     for (size_t i = idx - 1; blp == -1 && i != mu::nidx; --i) {
         const auto crst = elements[i];
         if (crst->isChord()) {
             blp = toChord(crst)->beams();
+        } else {
+            blp = toRest(crst)->beams();
```

Comment:
Since all methods that are being used (i.e. `beams`, `beamMode`) are actually ChordRest methods, there is not need to use `isChord`, `toChord`, `toRest`.
(Same below)

Reply:
Thanks for the hint, I simplified it. (I could have looked it up, but I trusted the existing code.)